src/test/java/com/wikia/webdriver/common/core/imageutilities/Shooter.java

```diff
@@ -42,38 +43,54 @@ public File capturePage(WebDriver driver) {
    * @param driver  - instance of WebDriver
    * @return File path  - file's handler which was saved in given path
    */
-  public File captureWebElement(WebElement element, WebDriver driver) {
-    Point start = element.getLocation();
-    Dimension size = element.getSize();
+
+  public BufferedImage takeScreenshot(WebElement element, WebDriver driver) {
+
+
+    JavascriptActions jsActions = new JavascriptActions();
+    CoordsProvider coordsProvider;
     if (!"FF".equals(Configuration.getBrowser())) {
-      Object[] rect = getBoundingClientRect(element, driver);
-      start = (Point) rect[0];
-      size = (Dimension) rect[1];
+      jsActions.scrollElementIntoViewPort(element);
+      coordsProvider = new SimpleViewportRelativeCoordsProvider();
+    } else {
+      coordsProvider = new WebDriverCoordsProvider();
     }
 
-    File image = imageEditor.cropImage(start, size, capturePage(driver));
-    PageObjectLogging.logImage("Shooter", image, true);
+    BufferedImage image = new AShot()
+            .coordsProvider(coordsProvider)
+            .shootingStrategy(ShootingStrategies.scaling(dpr))
+            .takeScreenshot(driver, element).getImage();
+
+    PageObjectLogging.log("Image size",image.getWidth() + "x" + image.getHeight(),true);
+    PageObjectLogging.logImage("Shooter", createTempFileFromImage(image), true);
     return image;
   }
 
-  private Object[] getBoundingClientRect(WebElement element, WebDriver driver) {
-    JavascriptExecutor js = (JavascriptExecutor) driver;
-    ArrayList<String> list = (ArrayList<String>) js.executeScript(
-        "var rect =  arguments[0].getBoundingClientRect();" +
-        "return [ '' + parseInt(rect.left), '' + parseInt(rect.top), '' + parseInt(rect.width), '' + parseInt(rect.height) ]",
-        element
-    );
-    Point start = new Point(Integer.parseInt(list.get(0)), Integer.parseInt(list.get(1)));
-    Dimension size = new Dimension(Integer.parseInt(list.get(2)), Integer.parseInt(list.get(3)));
-    return new Object[]{start, size};
-  }
+  private File createTempFileFromImage(BufferedImage image) {
+    File file;
+    try {
+      file = File.createTempFile("screenshot", ".png");
+    } catch (IOException e) {
+      throw new WebDriverException(e);
+    }
 
-  public BufferedImage takeScreenshot(WebElement element, WebDriver driver) {
-    return imageEditor.fileToImage(captureWebElement(element, driver));
+    try {
+      ImageIO.write(image, "png", file);
+    } catch (IOException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)